documentation/doxygen/makerootfile.C

```diff
@@ -0,0 +1,58 @@
+/// Generates the root file output of the macro MacroName
+
+
+
+void makerootfile(const char *MacroName, const char *IN, const char *OutDir, bool cp, bool py)
+{
+
+   // Execute the macro as a C++ one or a Python one.
+   if (!py) gROOT->ProcessLine(Form(".x %s",MacroName));
+   else     gROOT->ProcessLine(Form("TPython::ExecScript(\"%s\");",MacroName));
+
+   // If needed, copy the macro in the documentation directory.
+   if (cp) {
+      TString MN = MacroName;
+      Int_t i = MN.Index("(");
+      Int_t l = MN.Length();
+      if (i>0) MN.Remove(i, l);
+      gSystem->Exec(TString::Format("cp %s %s/macros", MN.Data(), OutDir));
+   }
+
+   // Iterate over all the pads generated by the macro and produce the corresponding pictures.
+   TIter iCanvas(gROOT->GetListOfCanvases());
+   TCanvas* canvas = 0;
+   int ImageNum = 0;
+
+   auto f = new TFile( TString::Format("%s/html/%s.root",OutDir,IN),"RECREATE");
+
+   int cw, ch;
+   while ((canvas = (TCanvas*) iCanvas())) {
+      ImageNum++;
+      canvas->SetName(TString::Format("pict%d_%s",ImageNum,IN));
+      cw = canvas->GetWindowWidth();
+      ch = canvas->GetWindowHeight();
+      canvas->Write();
+   }
+
+   f->Close();
+
+   // Build the html file inlining the JS pictures
+   int i;
+   FILE *fh = fopen(TString::Format("%s/macros/%s.html",OutDir,IN), "w");
+   fprintf(fh,"<script src=\"https://root.cern.ch/js/dev/scripts/JSRootCore.js\" type=\"text/javascript\"></script>\n");
+   fprintf(fh,"<script type='text/javascript'>\n");
+   fprintf(fh,"JSROOT.OpenFile(\"./%s.root\", function(file) {\n",IN);
+   for (i=1; i<=ImageNum; i++) {
+      fprintf(fh,"   file.ReadObject(\"pict%d_%s\", function(obj) {JSROOT.draw(\"draw_pict%d_%s\", obj, \"\");});\n",
+                 i,IN,i,IN);
+   }
+   fprintf(fh,"});\n");
+   fprintf(fh,"</script>\n");
+   fprintf(fh,"<center>\n");
+   for (i=1; i<=ImageNum; i++) {
+      fprintf(fh,"<div id=\"draw_pict%d_%s\" style=\"width:%dpx; height:%dpx\"></div>\n",
+              i,IN,cw,ch);
```

Comment:
Better if `<div>` element placed before `<script>` where div is used.
On some browsers this can lead to the errors.

Reply:
You mean we first to the <div> and then the <script> ? as the <div> is using the script I thought it was more logical to pout the div after... I will try. Thanks

Reply:
I have tried to invert div and script as follow:
```
<center>
<div id="draw_pict1_ContourList" style="width:678px; height:654px"></div>
<div id="draw_pict2_ContourList" style="width:678px; height:654px"></div>
</center>
<script type='text/javascript'>
JSROOT.OpenFile("./ContourList.root", function(file) {
   file.ReadObject("pict1_ContourList", function(obj) {JSROOT.draw("draw_pict1_ContourList", obj, "");});
   file.ReadObject("pict2_ContourList", function(obj) {JSROOT.draw("draw_pict2_ContourList", obj, "");});
});
</script>
```
On Mac with all the browsers I have have (Safari, Firefox, Chrome, Brave) it works fine in that order. Do you think I should do it this way ? (div first then script).

Reply:
> Do you think I should do it this way ? (div first then script).

Yes, this is right order.